src/rust/inetstack/protocols/layer3/arp/header.rs

```diff
@@ -121,9 +120,12 @@ impl ArpHeader {
         Ok(pdu)
     }
 
-    /// Serializes the target ARP PDU.
-    pub fn serialize(&self, buf: &mut [u8]) {
-        let buf: &mut [u8; ARP_MESSAGE_SIZE] = (&mut buf[..ARP_MESSAGE_SIZE]).try_into().unwrap();
+    /// Create an ARP message and return the packetized buffer.
+    pub fn serialize_and_attach(&self) -> DemiBuffer {
+        let mut pkt: DemiBuffer = DemiBuffer::new_with_headroom(0, (ARP_MESSAGE_SIZE + ETHERNET2_HEADER_SIZE) as u16);
```

Comment:
Since we are creating a brand new packet here, we need to know how much room to allocate for all of its headers, requiring a downward dependency to the ethernet layer.

Reply:
Do you think adding this as a comment in the code would be better/clearer? And we can remove it once this is no longer true.